### DISCUSSION TITLE
Global Custom Blocks

### DISCUSSION THREAD

**Discussion by halti3615** [ScratchAddons/ScratchAddons#4800] [Ideas]
This is something that has been annoying me, and possibly many others for a long time now. It is the fact that custom blocks can only be ran in the sprite in which the block is defined in. For example, we have with 30 sprites or so. If I want to create a custom block for all sprites, I need to duplicate the definitions into every single sprite, not to mention fixing bugs this way, is incredibly tedious. This issue has been brought up in a community post to the real scratch devs too, but so much time has passed without anything happening, that I thought it might be a good idea to bring it up here. I can also suggest to additionally create an option too, to run the custom blocks within each sprite, or just being able to access the custom blocks of another sprite. (For example,  if Sprite1, Sprite2 and Sprite3 are our sprites and Block1 is our custom block within Sprite 1, we should be able to choose to run Block1 in all sprites separately, or just run Block1 from Sprite2, and only Sprite1 will run the code.)

[https://scratch.mit.edu/discuss/topic/12015/](url)

Cheers!

**Comment by pufferfish101007** [comment]
#1267

**Reply by halti3615** [reply]
To me it seems like that thread is quite dead though... I think bringing it up could motivate someone with the necessary skills to try and implement it. Also, I feel like my idea is a bit more than issues highlighted in the link you sent.
